src/styles/bannerCommon.ts

```diff
@@ -1,240 +1,310 @@
-    wrapper: css`
-        box-sizing: border-box;
-        display: flex;
-        flex-direction: column;
-        justify-content: center;
-        align-items: center;
-        width: 100%;
-        background-color: ${bannerColor};
-        color: ${neutral[20]};
-
-        html {
-            box-sizing: border-box;
-        }
-
-        *,
-        *:before,
-        *:after {
-            box-sizing: inherit;
-        }
-    `,
-
-    contentContainer: css`
-        align-items: stretch;
-        display: flex;
-        flex-direction: column;
-        position: relative;
-        margin: 0 auto;
-        width: 100%;
-        max-width: 740px;
-
-        ${from.desktop} {
-            max-width: 980px;
-            flex-direction: row;
-            min-height: ${imgHeight}px;
-        }
-
-        ${from.leftCol} {
-            max-width: 1140px;
-        }
 
-        ${from.wide} {
-            max-width: 1300px;
-        }
-    `,
-
-    topLeftComponent: css`
-        width: 93%;
-        padding: ${space[4]}px;
-        min-height: ${imgHeight}px;
-        ${from.desktop} {
-            width: 60%;
-        }
-    `,
-
-    bottomRightComponent: css`
-        display: flex;
-        justify-content: center;
-        width: 100%;
-        max-height: 100%;
-
-        ${from.desktop} {
-            align-self: flex-end;
-            padding-right: ${space[4]}px;
-            max-width: 45%;
-            justify-content: flex-end;
-        }
-
-        ${from.leftCol} {
-            padding-right: 0;
-            justify-content: space-between;
+export interface StyleData {
+    styleBackground: string;
+    styleHeader: string;
+    styleBody: string;
+    styleHighlight: string;
+    styleHighlightBackground: string;
+    styleButton: string;
+    styleButtonBackground: string;
+    styleButtonHover: string;
+    styleClose: string;
+    styleCloseBackground: string;
+    styleCloseHover: string;
+}
+
+const colorStringStyles = [
+    'styleBackground',
+    'styleHeader',
+    'styleBody',
+    'styleHighlight',
+    'styleHighlightBackground',
+    'styleButton',
+    'styleButtonBackground',
+    'styleButtonHover',
+    'styleClose',
+    'styleCloseBackground',
+    'styleCloseHover',
+];
+
+type Styles = keyof StyleData;
+
+export const selfServeStyles = (userVals: Extras, defaults: StyleData) => {
+    const style: StyleData = Object.assign({}, defaults);
+    const defKeys: Styles[] = Object.keys(defaults) as Styles[];
+    const regex = new RegExp(/^#([A-Fa-f0-9]{6})$/);
+
+    defKeys.forEach((key) => {
+        const userVal = userVals[key];
+
+        // If user val is undefined, or an empty string, use default val
+        if (userVal != null && userVal.length > 0) {
+            // Protect against CSS injection
+            const item = userVal.split(';')[0].trim();
+
+            // Protect against null or empty user strings
+            if (item == null || !item.length) {
+                return;
```

Comment:
Super minor, but I wonder, since we're using an early `return` here, should we be consistent and do that elsewhere in this function instead of nested `if`s? I think I find it a bit more readable when one of those two approaches is used rather than mixing them.

Reply:
if statements un-nested, as suggested